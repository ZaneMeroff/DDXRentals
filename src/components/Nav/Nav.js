import React, { Component } from 'react';
import avatar from '../../data/ski-avatar.png'
import snowflake from '../../data/snowflake.png'
import logo from '../../data/DDR-logo.png'
import './Nav.scss';
import { Link } from 'react-router-dom';

class Nav extends Component {
  constructor(props) {
    super(props);
    this.state = {
    }
  }

  render() {
    return (
      <nav className='nav-bar'>
        <Link to={`/`}><img className='logo-img' src={ logo } alt='DDR logo' /></Link>
        <h3 className='welcome-msg'>{`Welcome ${this.props.userData.name}! Thanks for choose DDR for your ${this.props.userData.value} Adventure`}</h3>
        <div className='user-display'>
          <h3 className='user-info'>{`${this.props.userData.name}`}</h3>
          <h3 className='user-info'>{`${this.props.userData.email}`}</h3>
        </div>
        <img className='avatar-img user-info' src={ avatar } alt='winter avatar with beanie and goggles' />
        <button type='button' className='user-info' id='logout-btn'>Logout</button>
        <img className='snowflake user-info' src={ snowflake } alt='snowflake favorite icon' />
      </nav>
    )
  }
}

export default Nav;
