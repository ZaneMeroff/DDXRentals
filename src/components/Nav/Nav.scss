.avatar-img {
  height: 60px;
  width: auto;
}

.nav-bar {
  height: 80px;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  background-color: #DB49CB;
  position: fixed;
  align-items: center;
  color: black;
  font-size: 1.2em;
  padding-right: 15px;
  width: 100%;
}

.logo-img {
  height: 150px;
}

.nav-btn {
  height: 35px;
}

.user-info {
  text-align: center;
  font-size: 1em;
  color: #FFBF03;
}

#logout-btn {
  font-size: .8em;
  color: #DB49CB;
  background-color: #fff;
  width: 80px;
  border-radius: 15px;
  height: 30px;
}

#logout-btn:hover {
  cursor: pointer;
  background-color: #02A1FF;
  border: none;
  color: #FFF;
}

.email {
  font-size: 0.8em;
  color: #FFF;
}

.welcome-msg {
  width: 30%;
  text-align: center;
  font-size: 1em;
}

.user-display {
  display: flex;
  flex-direction: column;
}

.account-btn {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  font-size: 0.6em;
  color: #FFF;
}

.account-btn:hover {
  color: #FFBF03;
}
