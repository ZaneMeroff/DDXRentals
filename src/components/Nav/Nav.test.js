import React from 'react';
import ReactDOM from 'react-dom';
import Nav from './Nav';
import { shallow } from 'enzyme';

describe('Nav', () => {

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Nav
      {{userData={
        check: false,
        email: '',
        name: '',
        validateRouterLink: '',
        value: "business"
      }}}
      />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('should be an instance of Nav component', () => {
    const wrapper = shallow(<Nav

      />);
      expect(wrapper).toMatchSnapshot();
  });

});
