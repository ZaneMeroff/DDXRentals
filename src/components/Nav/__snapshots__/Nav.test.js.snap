// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav should be an instance of Nav component 1`] = `
<nav
  className="nav-bar"
>
  <Link
    to="/"
  >
    <img
      alt="DDR logo"
      className="logo-img"
      src="DDR-logo.png"
    />
  </Link>
  <h3
    className="welcome-msg"
  >
    Welcome  to your business Adventure
  </h3>
  <div
    className="user-display"
  >
    <h3
      className="user-info"
    />
    <h3
      className="user-info email"
    />
  </div>
  <Link
    to="/account"
  >
    <div
      className="account-btn"
    >
      <img
        alt="snowflake favorite icon"
        className="nav-btn user-info"
        src="snowflake.png"
      />
      <h3>
        Favorites
      </h3>
    </div>
  </Link>
  <Link
    to="/map"
  >
    <div
      className="account-btn"
    >
      <img
        alt="list icon"
        className="nav-btn user-info"
        src="map.png"
      />
      <h3>
        Neighborhood
      </h3>
    </div>
  </Link>
  <Link
    to="/listings"
  >
    <div
      className="account-btn"
    >
      <img
        alt="list icon"
        className="nav-btn user-info"
        src="list.png"
      />
      <h3>
        All Listings
      </h3>
    </div>
  </Link>
  <Link
    to="/"
  >
    <button
      className="user-info"
      id="logout-btn"
      type="button"
    >
      Logout
    </button>
  </Link>
</nav>
`;
